src/main/java/com/google/api/codegen/discovery/transformer/java/JavaMethodToViewTransformer.java

```diff
@@ -0,0 +1,164 @@
+import com.google.api.codegen.util.java.JavaTypeTable;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.protobuf.Method;
+
+/*
+ * Transforms a Method and SampleConfig into the standard discovery surface for
+ * Java.
+ */
+public class JavaMethodToViewTransformer implements MethodToViewTransformer {
+
+  private final static String TEMPLATE_FILENAME = "java/discovery_fragment.snip";
+
+  public JavaMethodToViewTransformer() {}
+
+  @Override
+  public ViewModel transform(Method method, SampleConfig sampleConfig) {
+    SampleTypeTable sampleTypeTable =
+        new SampleTypeTable(
+            new JavaTypeTable(), new JavaSampleTypeNameConverter(sampleConfig, method.getName()));
+    JavaSampleNamer namer = new JavaSampleNamer();
+    SampleTransformerContext context =
+        SampleTransformerContext.create(sampleConfig, sampleTypeTable, namer, method.getName());
+    return createSampleView(context);
+  }
+
+  private SampleView createSampleView(SampleTransformerContext context) {
+    addStaticImports(context);
+    SampleConfig sampleConfig = context.getSampleConfig();
+    SampleTypeTable typeTable = context.getTypeTable();
+
+    return SampleView.newBuilder()
+        .templateFileName(TEMPLATE_FILENAME)
+        .outputPath(context.getMethodName() + ".frag.java")
+        .apiTitle(sampleConfig.apiTitle())
+        .apiName(sampleConfig.apiName())
+        .apiVersion(sampleConfig.apiVersion())
+        .body(createSampleBody(context))
+        .imports(typeTable.getImports())
+        .build();
+  }
+
+  public SampleBodyView createSampleBody(SampleTransformerContext context) {
+    SampleConfig sampleConfig = context.getSampleConfig();
+    MethodInfo methodInfo = sampleConfig.methods().get(context.getMethodName());
+    SampleNamer sampleNamer = context.getNamer();
+    SampleTypeTable sampleTypeTable = context.getTypeTable();
+    SymbolTable symbolTable = new SymbolTable();
+
+    SampleBodyView.Builder sampleBodyView = SampleBodyView.newBuilder();
+    sampleBodyView.serviceVarName(symbolTable.getNewSymbol(sampleNamer.getServiceVarName()));
+    sampleBodyView.serviceTypeName(sampleTypeTable.getAndSaveNicknameFor(sampleConfig));
+    sampleBodyView.methodNameComponents(methodInfo.nameComponents());
+    sampleBodyView.requestVarName(symbolTable.getNewSymbol("request"));
+    sampleBodyView.requestTypeName(
+        sampleTypeTable.getAndSaveNicknameFor(methodInfo.requestType(), true));
+
+    sampleBodyView.requestBodyVarName("");
+    sampleBodyView.requestBodyTypeName("");
+    sampleBodyView.resourceGetterName("");
+    sampleBodyView.resourceTypeName("");
+    sampleBodyView.isResourceMap(false);
+
+    if (methodInfo.isPageStreaming()) {
+      FieldInfo fieldInfo = methodInfo.pageStreamingResourceField();
+      // TODO(garrettjones): Should I form this name this way?
+      sampleBodyView.resourceGetterName(Name.lowerCamel("get", fieldInfo.name()).toLowerCamel());
```

Comment:
The logic should be in SampleNamer. See for example: https://github.com/googleapis/toolkit/blob/master/src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java#L153